src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -934,6 +931,31 @@ private async Task CommitCoreAsync(IUIThreadOperationContext operationContext, b
         }
     }
 
+    private async Task<Solution> GetFinalSolutionAsync(ImmutableArray<(DocumentId, string, SyntaxNode, SourceText)> documentChanges)
+    {
+        // Grab the workspace's current solution, and make the document changes to it we computed.
+        var finalSolution = Workspace.CurrentSolution;
+        foreach (var (documentId, newName, newRoot, newText) in documentChanges)
+        {
+            if (documentId.IsSourceGenerated)
+            {
+                var document = await finalSolution.GetDocumentAsync(documentId, includeSourceGenerated: true, CancellationToken.None).ConfigureAwait(false);
+                Contract.ThrowIfFalse(document.IsRazorSourceGeneratedDocument());
+            }
+
+            finalSolution = newRoot != null
+                ? finalSolution.WithDocumentSyntaxRoot(documentId, newRoot)
+                : finalSolution.WithDocumentText(documentId, newText);
+
+            if (!documentId.IsSourceGenerated)
+            {
+                finalSolution = finalSolution.WithDocumentName(documentId, newName);
+            }
```

Comment:
could you even have a case where you had an SG doc, and it gota  new name?  is this just being paranoid (which i'm fine with).  might be good to doc.

Reply:
Source generated documents don't support modifying document attributes, which this call does. Just like `WithDocumentText` and `WithSyntaxRoot` weren't supported until I added it a few months ago.

It's a bit of an annoying chicken and egg - it would be nice if this call could no-op if the attributes haven't actually changed, regardless of document types, but it means adding support for source generated documents in order to get far enough to know whether that is what is happening.